### PR TITLE
Add objectives

### PR DESCRIPTION
**Pull request by fernandoalmeida**

### CODE REVIEW

File: elixir_buildpack.config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+elixir_version=1.10.1
```
Comment (adrianolisboa) [inline elixir_buildpack.config:1]:
```suggestion
elixir_version=1.10.2
```
The last one from february is `1.10.2`:
https://github.com/elixir-lang/elixir/releases/tag/v1.10.2

File: lib/didata/studies.ex [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  @doc """
+  Returns the list of objectives.
+
+  ## Examples
+
+      iex> list_objectives()
+      [%Objective{}, ...]
+
+  """
+  def list_objectives do
```
Comment (adrianolisboa) [inline lib/didata/studies.ex:20]:
```suggestion
  def list_all_objectives do
```

File: lib/didata/studies.ex [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  ## Examples
+
+      iex> get_objective!(123)
+      %Objective{}
+
+      iex> get_objective!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_objective!(id), do: Repo.get!(Objective, id)
```
Comment (adrianolisboa) [inline lib/didata/studies.ex:38]:
```suggestion
  def get_objective_by_id!(id), do: Repo.get!(Objective, id)
```

File: lib/didata/studies.ex [hunk 4]
```diff
@@ -0,0 +1,104 @@
+defmodule Didata.Studies do
```
Comment (adrianolisboa) [inline lib/didata/studies.ex:1]: I really don't have any suggestion but naming the context by `Studies` looks weird to me. What do you think?

File: lib/didata_web/router.ex [hunk 5]
```diff
@@ -17,6 +17,7 @@ defmodule DidataWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
```
Comment (adrianolisboa) [inline lib/didata_web/router.ex:19]: It does make sense to make it a versionated API?

File: mix.exs [hunk 6]
```diff
@@ -42,6 +42,7 @@ defmodule Didata.MixProject do
       {:phoenix_live_reload, "~> 1.2", only: :dev},
       {:gettext, "~> 0.11"},
       {:jason, "~> 1.0"},
+      {:distillery, "~> 2.1"},
```
Comment (adrianolisboa) [inline mix.exs:45]: 👏

File: elixir_buildpack.config (hunk 1 above)
Comment (fernandoalmeida) [reply elixir_buildpack.config:1]: nice! I will update with a separated PR :+1:

File: lib/didata/studies.ex (hunk 2 above)
Comment (fernandoalmeida) [reply lib/didata/studies.ex:20]: these names ware given by Phoenix generator, I will maintain to follow the standard ok?
PS: I did not like this file at all and probably we will refactor it after the MVP

File: lib/didata/studies.ex (hunk 3 above)
Comment (fernandoalmeida) [reply lib/didata/studies.ex:38]: these names ware given by Phoenix generator, I will maintain to follow the standard ok?
PS: I did not like this file at all and probably we will refactor it after the MVP

File: lib/didata/studies.ex (hunk 4 above)
Comment (fernandoalmeida) [reply lib/didata/studies.ex:1]: It's ok to me, I bilieve it is the core domain of the project (related to objectives, areas, disciplines, topics, etc), we may rename it later if we find a better name. what do you thing?

File: lib/didata_web/router.ex (hunk 5 above)
Comment (fernandoalmeida) [reply lib/didata_web/router.ex:19]: we don't have API in this first version, this is a route generated by Phoenix. I removed the API config.

File: lib/didata/studies.ex (hunk 4 above)
Comment (fernandoalmeida) [reply lib/didata/studies.ex:1]: I am going to merge but we can discuss about and change it with other PR if necessary, right?

**Pull request merged**